src/preferences/dialog/dlgprefautodj.cpp

```diff
@@ -186,3 +201,9 @@ void DlgPrefAutoDJ::slotToggleRandomQueue(int buttonState) {
     m_pConfig->setValue(ConfigKey("[Auto DJ]", "EnableRandomQueueBuff"),
             enable);
 }
+
+void DlgPrefAutoDJ::slotToggleXFaderCenterResetState(int buttonState) {
+    bool enable = buttonState == Qt::Checked;
+    m_pConfig->setValue(ConfigKey("[Auto DJ]", "XFaderCenterResetState"),
+            enable);
```

Comment:
This basically circumvents slotApply() and applies the selection right away.

All this "buffering" to the config is not good pratice as it unnnecessarily increases complexity and pollutes the config file with temp values. This page hasn't been touched for years. So essentially, if you want start working on (and learning about) the preferences, the AutoDJ page is not a good example.

IMO we should built upon this, but instead seize the opportunity to refactor this page.
We can either
1. * stick with these checkbox / spinbox slots and store the temp values in member variables
* use the temp values in slotApply()
3. (preferred IMO) simply read all checkboxes and spinboxes in slotApply() and write to the config

Do you want to work on this refactoring?

Reply:
I indeed tried to imitate the given code as much as possible. With a lot of trial and error.
I saw the not cleaned buff-occurences in the cfg-file and searched for a method avoiding those extra lines,
I struggled myself through the code :-)
I don't have a clue how old certain files are, but I can imagine there were different people adding their own part (like I did) puzzling out how it could be done, finding a way and making it all more complex.
I compaired other dialog settings files, and the structure is each time different, there is no consistency (not even about checkbox in front or after the text), making it hard for newbies.
I think it started with making a spinbox depending on a checkbox and ended with a lot of buffers and rules.

I am honoured with your question and I would like to improve this part. I just can't give a timeline, my students are studying these days, next week I start taking exams, so the spare time willl be less.

Reply:
Sure, I understand, no need to excuse, I think that's how most contributors start.

If you decide to do the refactoring, please start over with a new branch. Or reorder commits and force-puhs, as you like.